scripts/zbtestcrypt

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+
+#  zbtestcrypt - test encryption by decrypting and re-encrypting
+# 
+#  Adam Laurie <adam@aperturelabs.com>
+#  http://www.aperturelabs.com
+# 
+
+import sys
+import argparse
+
+from scapy.all import *
+from killerbee import *
+from killerbee.scapy_extensions import *
+
+if __name__ == '__main__':
+    # Command-line arguments
+    parser = argparse.ArgumentParser(description="zbtestcrypt: \
+        Decrypt, re-encrypt and optionally transmit packets for testing and tool development.")
+    parser.add_argument('-c', '--channel', action='store', type=int, default=11,
+        help='tx/rx on given channel (default 11)')
+    parser.add_argument('-D', action='store_true', dest='showdev',
+        help='list KillerBee devices')
+    parser.add_argument('-i', '--interface', action='store', type=str, default=None,
+        help='provide the USB ID or Serial Device Path to use that device')
+    parser.add_argument('-k', '--network_key', action='store', type=str, default=None,
+        help='provide the NWK_KEY in HEX')
+    parser.add_argument('-l', '--link_key', action='store', type=str, default=None,
+        help='provide the LINK_KEY in HEX (APS layer not yet implemented!)')
+    parser.add_argument('-r', '--pcapfile', action='store', default=None,
+        help='pcap file to test')
+    parser.add_argument('-R', '--dsnafile', action='store', default=None,
+        help='Daintree SNA file to test')
+    parser.add_argument('-s', '--sleep', action='store', type=float, default=1.0,
+        help='if tx, wait given seconds between packet injections (default 1.0)')
+    parser.add_argument('-t', '--transmit_both', action='store_true', dest='transmit_both',
+        help='Transmit original and re-encrypted packets for independant capture/analysis')
+    parser.add_argument('-T', '--transmit_crypted', action='store_true', dest='transmit_crypted',
+        help='Transmit only re-encrypted packets for independant capture/analysis')
+    args = parser.parse_args()
+
+    if args.showdev:
+        show_dev()
+        exit(False)
+
+    if args.channel == None:
+        print >>sys.stderr, "ERROR: Must specify a channel."
```

Comment:
This one can fold to a required=True flag on the parser on line 20, although as there is a default set this can be deleted as it is irrelevant.